File: sdk/src/main/java/cz/ackee/resizin/Crop.java
package cz.ackee.resizin;

/**
 * Types of crop avaliable at resizin.com
 */
public enum Crop {
    FILL("fill"),
    FIT("fit"),
    SCALE("scale"),
    FACE("face"),
    PAD("pad");

    private final String value;

    Crop(String value) {
        this.value = value;
    }

    public String toString() {
        return this.value;
    }

    public static Crop fromValue(String value) {
        for (Crop c : values()) {
            if (c.value.equals(value)) {
                return c;
            }
        }
        throw new IllegalArgumentException("Invalid value " + value);
    }
}

File: sdk/src/main/java/io/github/ackeecz/resizin/sdk/Crop.kt
package io.github.ackeecz.resizin.sdk

/**
 * Types of crop avaliable at resizin.com
 */
enum class Crop(private val value: String) {

    FILL("fill"),
    FIT("fit"),
    SCALE("scale"),
    FACE("face"),
    PAD("pad");

    override fun toString() = value

    companion object {

        fun fromValue(value: String): Crop {
            return values().firstOrNull { it.value == value } ?: throw IllegalArgumentException("Invalid value $value")
        }
    }
}
